mne/io/base.py

```diff
@@ -33,17 +33,156 @@
 from ..event import concatenate_events
 
 
+class ToDataFrameMixin(object):
+    '''Class to add to_data_frame capabilities to certain classes.'''
+    def to_data_frame(self, picks=None, index=None, scale_time=1e3,
+                      scalings=None, copy=True, start=None, stop=None):
+        """Get the epochs as Pandas DataFrame
+
+        Export epochs data in tabular structure with MEG channels as columns
+        and three additional info columns 'epoch', 'condition', and 'time'.
+        The format matches a long table format commonly used to represent
+        repeated measures in within-subject designs.
+
+        Parameters
+        ----------
+        picks : array-like of int | None
+            If None only MEG and EEG channels are kept
+            otherwise the channels indices in picks are kept.
+        index : tuple of str | None
+            Column to be used as index for the data. Valid string options
+            are 'epoch', 'time' and 'condition'. If None, all three info
+            columns will be included in the table as categorial data.
+        scale_time : float
+            Scaling to be applied to time units.
+        scalings : dict | None
+            Scaling to be applied to the channels picked. If None, defaults to
+            ``scalings=dict(eeg=1e6, grad=1e13, mag=1e15, misc=1.0)`.
+        copy : bool
+            If true, data will be copied. Else data may be modified in place.
+        start, stop : int | None
+            If it is a Raw object, this defines a start/stop index for creating
+            the dataframe from a slice. The times will be interpolated from the
+            index and the sampling rate of the signal.
+
+        Returns
+        -------
+        df : instance of pandas.core.DataFrame
+            A dataframe suitable for usage with other
+            statistical/plotting/analysis packages. Column/Index values will
+            depend on the object type being converted, but should be
+            human-readable.
+        """
+        from ..epochs import Epochs, EpochsArray
+        from ..evoked import Evoked, EvokedArray
+        from .fiff import RawFIFF
+        from .array import RawArray
+        from ..source_estimate import SourceEstimate
+
+        pd = _check_pandas_installed()
+
+        ####### ADDITION #######
+        if isinstance(self, (Epochs, EpochsArray)):
+            default_index = ['condition', 'epoch', 'time']
+        elif isinstance(self, (RawFIFF, RawArray, Evoked,
+                               EvokedArray, SourceEstimate)):
+            default_index = ['time']
+
+        if index is not None:
+            _check_pandas_index_arguments(index, default_index)
+        else:
+            index = default_index
+
+        if picks is None:
+            picks = list(range(self.info['nchan']))
+        else:
+            if isinstance(self, (Epochs, EpochsArray)):
+                if not in1d(picks, np.arange(len(self.events))).all():
```

Comment:
I don't think this line is correct anyway (which is why I was confused during re-coding). `picks` has to do with the number of channels, `self.events` has to do with the number of epochs... I'm not sure why you'd compare them.